src/USER-MDI/mdi_engine.cpp

```diff
@@ -0,0 +1,364 @@
+  MDI_Register_Node("@PRE-FORCES");
+  MDI_Register_Command("@PRE-FORCES", "<@");
+  MDI_Register_Command("@PRE-FORCES", "<CELL");
+  MDI_Register_Command("@PRE-FORCES", "<CHARGES");
+  MDI_Register_Command("@PRE-FORCES", "<COORDS");
+  MDI_Register_Command("@PRE-FORCES", "<ENERGY");
+  MDI_Register_Command("@PRE-FORCES", "<FORCES");
+  MDI_Register_Command("@PRE-FORCES", "<KE");
+  MDI_Register_Command("@PRE-FORCES", "<LABELS");
+  MDI_Register_Command("@PRE-FORCES", "<MASSES");
+  MDI_Register_Command("@PRE-FORCES", "<NATOMS");
+  MDI_Register_Command("@PRE-FORCES", "<PE");
+  MDI_Register_Command("@PRE-FORCES", ">COORDS");
+  MDI_Register_Command("@PRE-FORCES", ">FORCES");
+  MDI_Register_Command("@PRE-FORCES", "@");
+  MDI_Register_Command("@PRE-FORCES", "@COORDS");
+  MDI_Register_Command("@PRE-FORCES", "@DEFAULT");
+  MDI_Register_Command("@PRE-FORCES", "@FORCES");
+  MDI_Register_Command("@PRE-FORCES", "@PRE-FORCES");
+  MDI_Register_Command("@PRE-FORCES", "EXIT");
+
+  // register the forces node
+  MDI_Register_Node("@FORCES");
+  MDI_Register_Callback("@FORCES", ">FORCES");
+  MDI_Register_Command("@FORCES", "<@");
+  MDI_Register_Command("@FORCES", "<CELL");
+  MDI_Register_Command("@FORCES", "<CHARGES");
+  MDI_Register_Command("@FORCES", "<COORDS");
+  MDI_Register_Command("@FORCES", "<ENERGY");
+  MDI_Register_Command("@FORCES", "<FORCES");
+  MDI_Register_Command("@FORCES", "<KE");
+  MDI_Register_Command("@FORCES", "<LABELS");
+  MDI_Register_Command("@FORCES", "<MASSES");
+  MDI_Register_Command("@FORCES", "<NATOMS");
+  MDI_Register_Command("@FORCES", "<PE");
+  MDI_Register_Command("@FORCES", ">COORDS");
+  MDI_Register_Command("@FORCES", ">FORCES");
+  MDI_Register_Command("@FORCES", "@");
+  MDI_Register_Command("@FORCES", "@COORDS");
+  MDI_Register_Command("@FORCES", "@DEFAULT");
+  MDI_Register_Command("@FORCES", "@FORCES");
+  MDI_Register_Command("@FORCES", "@PRE-FORCES");
+  MDI_Register_Command("@FORCES", "EXIT");
+
+  // register the coordinates node
+  MDI_Register_Node("@COORDS");
+  MDI_Register_Command("@COORDS", "<@");
+  MDI_Register_Command("@COORDS", "<CELL");
+  MDI_Register_Command("@COORDS", "<CHARGES");
+  MDI_Register_Command("@COORDS", "<COORDS");
+  MDI_Register_Command("@COORDS", "<ENERGY");
+  MDI_Register_Command("@COORDS", "<FORCES");
+  MDI_Register_Command("@COORDS", "<KE");
+  MDI_Register_Command("@COORDS", "<LABELS");
+  MDI_Register_Command("@COORDS", "<MASSES");
+  MDI_Register_Command("@COORDS", "<NATOMS");
+  MDI_Register_Command("@COORDS", "<PE");
+  MDI_Register_Command("@COORDS", ">COORDS");
+  MDI_Register_Command("@COORDS", ">FORCES");
+  MDI_Register_Command("@COORDS", "@");
+  MDI_Register_Command("@COORDS", "@COORDS");
+  MDI_Register_Command("@COORDS", "@DEFAULT");
+  MDI_Register_Command("@COORDS", "@FORCES");
+  MDI_Register_Command("@COORDS", "@PRE-FORCES");
+  MDI_Register_Command("@COORDS", "EXIT");
+
+
+  // identify the mdi_engine fix
+  bool found_fix = false;
```

Comment:
why not use `modify->find_fix()`?

Reply:
Yes, `modify->find_fix_by_style()` is better here.  I've made the change.